nanoFramework.SignalR.Client/AsyncResult.cs

```diff
@@ -97,23 +97,15 @@ internal void Setvalue(object contentValue)
             if (contentValue != null)
             {
                 string[] fullNames = ReturnType.FullName.Split('.');
-                if (fullNames.Length == 2 && fullNames[0] == "System")
-                {
-                    //base types can be cast directly
-                    _value = contentValue;
-                }
-                else
-                {
-                    _value = JsonConvert.DeserializeObject(JsonConvert.SerializeObject(contentValue), ReturnType);
-                }
+                _value = JsonConvert.DeserializeObject(JsonConvert.SerializeObject(contentValue), ReturnType);
```

Comment:
By curiosity, why have you decided to change?

Reply:
José changed the JSON for dealing with base types making this check obsolete. I now also removed the above line that split the Type name.

Reply:
then!